### PR TITLE
[TESTMERGE ONLY]MouseMove/MouseDrag testmerge PR

### PR DESCRIPTION
**Pull request by silicons**
@AnturK 
Also for the PR this is for, I'd rather not use MouseDrag as it only updates when you change tiles which is basically worthless as it makes pixel projectile aiming absolutely trash.

[For people who weren't in coderbus when we were discussing which is alot of you, this is to see if #27618 is going to fuck up the network connection of the server with its mouse tracking which is apparently a shitload of overhead..]

### CODE REVIEW

File: code/_onclick/autoclick.dm [hunk 1]
```diff
@@ -1,5 +1,10 @@
 /client
 	var/list/atom/selected_target[2]
+	var/obj/item/active_mousedown_item = null
+	var/mouseParams = ""
+	var/mouseLocation = null
+	var/mouseObject = null
+	var/mouseControlObject = null
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Remove the `= null`s

File: code/_onclick/autoclick.dm [hunk 2]
```diff
@@ -27,8 +44,31 @@
 	if(h)
 		. = h.CanItemAutoclick(object, location, params)
 
+/mob/proc/canMobMousedown(object, location, params)
+
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Add a return here

File: code/_onclick/autoclick.dm [hunk 3]
```diff
@@ -27,8 +44,31 @@
 	if(h)
 		. = h.CanItemAutoclick(object, location, params)
 
+/mob/proc/canMobMousedown(object, location, params)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of object and location

File: code/_onclick/autoclick.dm [hunk 4]
```diff
@@ -27,8 +44,31 @@
...
+
+/mob/living/carbon/canMobMousedown(atom/object, location, params)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of location

File: code/_onclick/autoclick.dm [hunk 5]
```diff
@@ -27,8 +44,31 @@
...
+	var/obj/item/H = get_active_held_item()
+	if(H)
+		. = H.canItemMouseDown(object, location, params)
+
 /obj/item/proc/CanItemAutoclick(object, location, params)
 
+/obj/item/proc/canItemMouseDown(object, location, params)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of object and location

File: code/_onclick/autoclick.dm [hunk 6]
```diff
@@ -27,8 +44,31 @@
...
+	if(canMouseDown)
+		return src
+
+/obj/item/proc/onMouseDown(object, location, params, mob)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of object and location. Rename mob to mob/M

File: code/_onclick/autoclick.dm [hunk 7]
```diff
@@ -27,8 +44,31 @@
...
+	return
+
+/obj/item/proc/onMouseUp(object, location, params, mob)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of object and location. Rename mob to mob/M

File: code/_onclick/autoclick.dm [hunk 8]
```diff
@@ -27,8 +44,31 @@
...
+	return
+
+/obj/item/proc/onMouseDrag(src_object, over_object, src_location, over_location, params, mob)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of src_object, over_object, src_location, and over_location. Rename mob to mob/M

File: code/_onclick/autoclick.dm [hunk 9]
```diff
@@ -27,8 +44,31 @@
...
+	return
+
+/obj/item
+	var/canMouseDown = FALSE
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Put this in items.dm

File: code/_onclick/autoclick.dm [hunk 10]
```diff
@@ -42,4 +82,11 @@
 	. = 0
 
 /obj/screen/click_catcher/IsAutoclickable()
-	. = 1
\ No newline at end of file
+	. = 1
+
+//Please don't roast me too hard
+/client/MouseMove(object,location,control,params)
```
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the type of object and location

File: code/_onclick/autoclick.dm (hunk 1 above)
Comment (Cyberboss) [inline code/_onclick/autoclick.dm]: Specify the types of mouseLocation, mouseObject, and mouseControlObject

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by Razharas** [comment]
1 - hardlocking it into items is not nice
2 - this whole autoclicker thingie can use a subsystem instead

**Comment by kevinz000** [comment]
@Razharas what else can you possibly need to mousedrag that can be held..?

**Comment by kevinz000** [comment]
right can someone please testmerge this at highpop so I can see if my other PR is viable in its current state?

**Comment by MrStonedOne** [comment]
2.8125kb/s per client to send MouseMove events (worst case. there is some over estimation on the packet size.)

**Comment by kevinz000** [comment]
@MrStonedOne bits or bytes and so in high-extreme pop(100) there's going to be 0.2 mb/s?
is that bad?

**Pull request closed without merging**